lib/net/ip.go

```diff
@@ -42,6 +42,12 @@ func (i *IP) UnmarshalJSON(b []byte) error {
 	return i.UnmarshalText([]byte(s))
 }
 
+// ParseIP returns an IP from a string
```

Comment:
The base call will return `nil` if it is invalid, I could do a nil check and throw an error so the problem doesn't make it further down the pipe before being caught, WDYT?